pkg/tsdb/prometheus/buffered/client.go

```diff
@@ -0,0 +1,80 @@
+package buffered
```

Comment:
Mostly taken from provider.go but split into separate functions so that HTTP roundtripper and the prom client can be created separately.